core/src/main/java/hudson/model/Queue.java

```diff
@@ -1460,54 +1460,64 @@ public void maintain() {
             // allocate buildable jobs to executors
             for (BuildableItem p : new ArrayList<BuildableItem>(
                     buildables)) {// copy as we'll mutate the list in the loop
-                // one last check to make sure this build is not blocked.
-                if (isBuildBlocked(p)) {
-                    p.leave(this);
-                    new BlockedItem(p).enter(this);
-                    LOGGER.log(Level.FINE, "Catching that {0} is blocked in the last minute", p);
-                    // JENKINS-28926 we have moved an unblocked task into the blocked state, update snapshot
-                    // so that other buildables which might have been blocked by this can see the state change
-                    updateSnapshot();
-                    continue;
+                if (p.task instanceof FlyweightTask) {
+                    Runnable r = makeFlyWeightTaskBuildable(new BuildableItem(p));
+                    if (r != null) {
+                        p.leave(this);
+                        r.run();
+                        updateSnapshot();
+                    }
                 }
+                else {
```

Comment:
please put this else on the same line as the closing `}` from the `if` statement that it belongs to